source/distributed/solution_transfer.cc

```diff
@@ -242,20 +244,45 @@ namespace parallel
             &dof_handler->get_triangulation())));
       Assert(tria != nullptr, ExcInternalError());
 
+      for (const auto vec : all_out)
+        *vec = 0.0;
```

Comment:
@peterrum For my code this line is problematic. I was relying on the fact that SolutionTransfer is not modifying entries that were previously associated with `FE_Nothing` so that I can set initial conditions before calling SolutionTransfer::interpolate.